static/src/javascripts/projects/common/modules/commercial/sticky-top-banner.js

```diff
@@ -46,26 +48,17 @@ define([
     // Rubicon ads are loaded via DFP like all other ads, but they can
     // render themselves again at any time
     var newRubiconAdHeightPromise = new Promise(function (resolve) {
-        window.addEventListener('message', function (event) {
-            var data;
-            // other DFP events get caught by this listener, but if they're not json we don't want to parse them or use them
-            try {
-                data = JSON.parse(event.data);
-            } catch (e) {/**/}
-
-            if (data) {
-                var $iframe = getAdIframe();
-                var isRubiconAdEvent = data.type === 'set-ad-height';
-                var isEventForTopAdBanner = isRubiconAdEvent && data.value.id === $iframe[0].id;
-
-                if (isRubiconAdEvent && isEventForTopAdBanner) {
-                    fastdom.read(function () {
-                        var padding = parseInt($adBannerInner.css('padding-top'))
-                            + parseInt($adBannerInner.css('padding-bottom'));
-                        var clientHeight = parseInt(data.value.height) + padding;
-                        resolve(clientHeight);
-                    });
-                }
+        messenger.register('set-ad-height', function onSetAdHeight(data) {
+            var $iframe = getAdIframe();
```

Comment:
hey @jbreckmckye ! Good to hear from you 😃  Indeed that is something I started thinking about too. I wrote this on the condition that our ads are running on the same domain than the page they're embedded in. From there, inside the frame you have access to `window.frameElement`. From the parent page, you can loop over `window.frames`. But there are other areas I am exploring:

- apparently the `name` attribute on the `iframe` element is passed down in the context of the embedded document, but I need to test this assertion
- DFP assigns `id`s to all those iframes (which is what @OliverJAsh used here)
- this module is intended to be used only by our in-house templates and so we can imagine some kind of shared secret (would also be useful to authenticate the origin of a message)